### ISSUE TITLE
funnel webpage does not show any tasks

### ISSUE THREAD

**Issue by kmavrommatis** [ohsu-comp-bio/funnel#609]
Hi,

The main page of the GUI does not show any tasks, in any state. This is a new behavior, the same setup was working few days back. Since then there has not been any change in the networking/infrastructure AFAIK.
I have tried  `funnel 0.9.0` as well as the version mentioned in #608.
The backend database is a mongodb (v 4.0.6) which has the table tasks populated and jobs are executed using AWS batch.
When I submit new jobs to the server they do get in the tasks list, and they are executed. They are also counted in the `Service Info` tab.

I have tried to use different browsers (Safari, Chrome, Firefox) and different computers, either local or AWS instances, with the same outcome.
Looking in the logs of funnel (captured in debug mode), when I refresh the page I can see messages like the below, which - at least to me - appear to indicate that the information is received from the database.

```
{"level":"debug","msg":"received: /tes.TaskService/GetTask","ns":"server","request":{"id":"bj2t68q6qkdg00fnmhp0","view":1},"time":"2019-04-30T23:56:27Z"}
{"level":"debug","msg":"received: /tes.TaskService/GetTask","ns":"server","request":{"id":"bj2t68q6qkdg00fnmhog","view":1},"time":"2019-04-30T23:56:27Z"}
{"err":null,"level":"debug","msg":"responding: /tes.TaskService/GetTask","ns":"server","resp":{"id":"bj2t68q6qkdg00fnmhp0","state":5,"name":"root.mergeFastq2","inputs":[{"name":"220619831_N_S24_L007_R2_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L007_R2_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L007_R2_001.fastq.gz"},{"name":"220619831_N_S24_L006_R2_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L006_R2_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L006_R2_001.fastq.gz"},{"name":"220619831_N_S24_L008_R2_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L008_R2_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L008_R2_001.fastq.gz"},{"name":"220619831_N_S24_L005_R2_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L005_R2_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L005_R2_001.fastq.gz"}],"outputs":[{"name":"mergeFastq2","url":"s3://s3.us-east-1.amazonaws.com/databucket/funnel_1/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2","path":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2","type":1}],"resources":{"cpu_cores":8},"executors":[{"image":"064965165932.dkr.ecr.us-east-1.amazonaws.com/ngs/systemtools:latest","command":["/bin/sh","-c","zcat  /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L005_R2_001.fastq.gz /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L006_R2_001.fastq.gz /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L007_R2_001.fastq.gz /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L008_R2_001.fastq.gz  | pigz -p 8 -   \u003e 0351003.1_R2.fq.gz"],"workdir":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2","stderr":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2/job.err.log","env":{"HOME":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2","TMPDIR":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2"}}],"tags":{"job_id":"f0ab0e70-445d-3f6b-9dec-8d85e03e298b","tool_name":"root.mergeFastq2","workflow_id":"7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e"},"logs":[{"logs":[{"start_time":"2019-04-28T16:39:44.716122195Z","end_time":"2019-04-28T16:45:51.687693188Z"}],"metadata":{"awsbatch_id":"f4471840-6474-4e5b-b6d9-d884361e1bb3","hostname":"ip-10-112-30-182"},"start_time":"2019-04-28T16:34:06.122681765Z","end_time":"2019-04-28T16:46:32.673737247Z","outputs":[{"url":"s3://s3.us-east-1.amazonaws.com/databucket/funnel_1/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2/0351003.1_R2.fq.gz","path":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2/0351003.1_R2.fq.gz","size_bytes":2721648331},{"url":"s3://s3.us-east-1.amazonaws.com/databucket/funnel_1/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2/job.err.log","path":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq2/job.err.log"}]}],"creation_time":"2019-04-28T16:21:55.159400363Z"},"time":"2019-04-30T23:56:27Z"}
{"err":null,"level":"debug","msg":"responding: /tes.TaskService/GetTask","ns":"server","resp":{"id":"bj2t68q6qkdg00fnmhog","state":5,"name":"root.mergeFastq1","inputs":[{"name":"220619831_N_S24_L006_R1_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L006_R1_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L006_R1_001.fastq.gz"},{"name":"220619831_N_S24_L005_R1_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L005_R1_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L005_R1_001.fastq.gz"},{"name":"220619831_N_S24_L008_R1_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L008_R1_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L008_R1_001.fastq.gz"},{"name":"220619831_N_S24_L007_R1_001.fastq.gz","url":"s3://s3.us-east-1.amazonaws.com/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L007_R1_001.fastq.gz","path":"/databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L007_R1_001.fastq.gz"}],"outputs":[{"name":"mergeFastq1","url":"s3://s3.us-east-1.amazonaws.com/databucket/funnel_1/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1","path":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1","type":1}],"resources":{"cpu_cores":8},"executors":[{"image":"064965165932.dkr.ecr.us-east-1.amazonaws.com/ngs/systemtools:latest","command":["/bin/sh","-c","zcat  /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L005_R1_001.fastq.gz /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L006_R1_001.fastq.gz /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L007_R1_001.fastq.gz /databucket/RNA-Seq/Raw/pilot/fastq.20190418/220619831_N_S24_L008_R1_001.fastq.gz  | pigz -p 8 -   \u003e 0351003.1_R1.fq.gz"],"workdir":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1","stderr":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1/job.err.log","env":{"HOME":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1","TMPDIR":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1"}}],"tags":{"job_id":"5e8dba05-ed7b-37e2-9138-86ffecb44b3a","tool_name":"root.mergeFastq1","workflow_id":"7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e"},"logs":[{"logs":[{"start_time":"2019-04-28T16:28:23.524548165Z","end_time":"2019-04-28T16:34:08.730394611Z"}],"metadata":{"awsbatch_id":"d05dfa97-0981-4af4-a9a1-81112a2174c0","hostname":"ip-10-112-31-48"},"start_time":"2019-04-28T16:23:26.883289327Z","end_time":"2019-04-28T16:34:42.957780927Z","outputs":[{"url":"s3://s3.us-east-1.amazonaws.com/databucket/funnel_1/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1/0351003.1_R1.fq.gz","path":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1/0351003.1_R1.fq.gz","size_bytes":2457631383},{"url":"s3://s3.us-east-1.amazonaws.com/databucket/funnel_1/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1/job.err.log","path":"/opt/funnel/7b7afab9-551c-4eb2-a1b1-7e8343fb3d2e/root/mergeFastq1/job.err.log"}]}],"creation_time":"2019-04-28T16:21:55.067238808Z"},"time":"2019-04-30T23:56:27Z"}
```

I have tried to restart the server, the instance it is running on and the mongodb without any change.

Any advise on how to address or further debug this issue?
Thanks in advance for your help

**Comment by adamstruck** [comment]
If you open up the developer tools in Chrome / Firefox do you see any errors in the javascript console?

**Comment by kmavrommatis** [comment]
Hi, I tried the Chrome Devtools and I could not see any obvious error.

The Chrome profiler indicates that initially time spent in Parsing the HTML for tasks and setting up the page but then it stays idle until the page is shown.

After trying several times today I finally got the list of tasks but it still takes long to load (>30s in some cases). This is consistent in both version of funnel that I am using.

**Comment by adamstruck** [comment]
Hi sorry for the delay.  Can you try the branch `react-webdash`? I've been working on the side to update the web dashboard.

**Comment by kmavrommatis** [comment]
Thanks, I like the new look of it on the front page. The details of a task are shown the JSON  contents directly, is this intentional?

**Comment by adamstruck** [comment]
It was originally a shortcut to get a prototype of dashboard up and running. I don't mind it personally, but I'd welcome any feedback.  I am not much of a frontend guy so this has all been new to me.

**Comment by kmavrommatis** [comment]
my 5c:
I think that new skin looks nice, and the page buttons are great. The way the content was organized in the task page in the previous version is more easily digestible by an end user (dealing with nested {} may be too much an ask for an average user)

**Comment by adamstruck** [comment]
fixed by #610
